src/kbmod/standardizers/butler_standardizer.py

```diff
@@ -0,0 +1,265 @@
+        # obvious for the end user. Optionally, we can give a mandatory
+        # **kwarg to every __init__ and rely on column naming and unpacking.
+        # this is not the best, explicit better than implicit, but also now
+        # the __init__ method kwargs need to be named the same as columns and
+        # be optional too. For example does this init make sense to you?
+        # I don't think I would be able to write this without knowing how the
+        # whole thing works...
+        #breakpoint()
+        self.butler = butler
+        self.location = butler.datastore.root
+        if datasetRefs is None:
+            # import this here for now, for optimization
+            from lsst.daf.butler.core import DatasetId
+            self.refs = [butler.registry.getDataset(DatasetId(id)),]
+        else:
+            self.refs = datasetRefs
+        # wth? why is it a data reference if it doesn't reference data without
+        # being explicit about it? Why `collection` and `collections` don't
+        # behave the same way? Why is this a thing! AAARRRGGGHH!
+        self.exts = [butler.get(ref, collections=[ref.run, ]) for ref in self.refs]
+        self._wcs = []
+        self._bbox = []
+
+    @property
+    def processable(self):
+        return self.exts
+
+    @property
+    def wcs(self):
+        if not self._wcs:
+            self._wcs = list(self.standardizeWCS())
+        return self._wcs
+
+    @property
+    def bbox(self):
+        if not self._bbox:
+            self._bbox = list(self.standardizeBBox())
+        return self._bbox
+
+    def _computeBBox(self, wcs, dimX, dimY):
+        """Given an WCS and the dimensions of an image calculates the values of
+        world coordinates at image corner and image center.
+
+        Parameters
+        ----------
+        wcs : `object`
+            The header, Astropy HDU and its derivatives.
+        dimX : `int`
+            Image dimension in x-axis.
+        dimY : `int`
+            Image dimension in y-axis.
+
+        Returns
+        -------
+        standardizedBBox : `dict`
+            Calculated coorinate values, a dict with, ``wcs_center_[ra, dec]``
+            and ``wcs_corner_[ra, dec]`` keys.
+
+        Notes
+        -----
+        The center point is assumed to be at the (dimX/2, dimY/2) pixel
+        coordinates, rounded down. Corner is taken to be the (0,0)-th pixel.
+        """
+        standardizedBBox = {}
+        centerX, centerY = int(dimX/2), int(dimY/2)
+
+        centerSkyCoord = wcs.pixel_to_world(centerX, centerY)
+        cornerSkyCoord = wcs.pixel_to_world(0, 0)
+
+        standardizedBBox["center_ra"] = centerSkyCoord.ra.deg
+        standardizedBBox["center_dec"] = centerSkyCoord.dec.deg
+
+        standardizedBBox["corner_ra"] = cornerSkyCoord.ra.deg
+        standardizedBBox["corner_dec"] = cornerSkyCoord.dec.deg
+
+        return standardizedBBox
+
+    def _maskSingleExp(self, exp):
```

Comment:
I really like how the structure works for preparing images. I think masking is a great example of the power of this approach.